src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -189,6 +208,21 @@ public static BinaryData FromObjectAsJson<T>(T jsonSerializable, JsonSerializerO
             return new BinaryData(buffer);
         }
 
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance by serializing the provided object using
+        /// the <see cref="JsonSerializer"/>.
+        /// </summary>
+        /// <typeparam name="T">The type to use when serializing the data.</typeparam>
+        /// <param name="jsonSerializable">The data to use.</param>
+        /// <param name="jsonTypeInfo">The <see cref="JsonTypeInfo"/> to use when serializing to JSON.</param>
+        /// <returns>A value representing the UTF-8 encoding of the JSON representation of <paramref name="jsonSerializable" />.</returns>
+        [RequiresUnreferencedCode(JsonSerializerRequiresUnreferencedCode)]
+        public static BinaryData FromObjectAsJson<T>(T jsonSerializable, JsonTypeInfo<T> jsonTypeInfo)
+        {
+            byte[] buffer = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, jsonTypeInfo);
+            return new BinaryData(buffer);
+        }
```

Comment:
subjective (feel free to ignore): we could use expression syntax here

```suggestion
            => new BinaryData(JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, jsonTypeInfo));
```